### PR TITLE
Implement robot tests, other stuff

### LINKED ISSUE #166: Implement robot integration tests
Using `robot.py test` still gives
```
tests are not yet implemented for RobotPy 2020
##[error]Process completed with exit code 1.
```

### PR DESCRIPTION
**Pull request by virtuald**
* Fixes #166 
* Requires https://github.com/robotpy/robotpy-hal/pull/16
* Requires https://github.com/robotpy/robotpy-wpilib/pull/704
* Requires https://github.com/wpilibsuite/allwpilib/pull/4007

### CODE REVIEW

File: pyfrc/physics/core.py [hunk 1]
```diff
@@ -88,14 +86,18 @@ class PhysicsInterface:
     """
 
     @classmethod
-    def _create(cls, robot_path):
+    def _create_and_attach(
+        cls: type["PhysicsInterface"],
+        robot_class: type[wpilib.RobotBase],
```
Comment (auscompgeek) [inline pyfrc/physics/core.py:91]: We'd need either `from __future__ import annotations` or use typing.Type instead.

**Comment by virtuald** [comment]
There's still a bunch of obnoxious global state that will still need to be dealt with. Unfortunately, I don't think any of them are fixable except by modifications to the C++ code.

* Command framework scheduler 
* SendableRegistry needs to be cleared
* Probably LiveWindow too

**Comment by virtuald** [comment]
https://github.com/wpilibsuite/allwpilib/pull/4007 is merged so that clears the way for pyfrc testing to happen if we want to use a development release until the next release happens (affects wpiutil, wpilib)... but I'm going away this weekend, so probably a bad idea to do it before next week.

**Comment by virtuald** [comment]
All of the vendor code I've tested does bad things when confronted with basic tests of creating/destroying objects in the same process:

* navx segfaults https://github.com/kauailabs/navxmxp/issues/104
* ctre segfaults
* rev hangs the test process on exit

**Comment by virtuald** [comment]
Due to potential issues with this, I'm not going to enable this for deploy by default

**Pull request merged**